datagen/generate.py

```diff
@@ -247,8 +295,28 @@ def generate_data_file():
                 "current": current_service_regime,
             },
         }
+    total_ridership_time_series = [sum(entries_for_day) for entries_for_day in zip(*ridership_time_series_list)]
```

Comment:
I created a function to be used under the `for` loop because what I have in the for loop depends on the existing contents and if I were to use a different function, I would have to duplicate the code from the` for` loop which didn't really make sense for me to do (unless there is a different approach I am unaware of). With that being said, for my function I needed 6 parameters, which feels like a lot. I'm not sure if this is good practice to have so many parameters or if there was a different way I could go about making the function.

```
def generate_total_data(ridership_time_series_list, service_time_series_list, combined_total_trips, total_cancelled_routes, total_reduced_serv_routes, total_increased_serv_routes):
    total_ridership_time_series = [sum(entries_for_day) for entries_for_day in zip(*ridership_time_series_list)] 
    condensed_ridership_series = condensed_time_series(total_ridership_time_series) 
    total_service_time_series = [sum(entries_for_day) for entries_for_day in zip(*service_time_series_list)] 
    condensed_service_series = condensed_time_series(total_service_time_series) 
    total_ridership_percentage = get_ridership_percentage(total_ridership_time_series) 
    total_service_percentage = get_service_percentage(total_service_time_series) 
    total_passengers = sum(total_ridership_time_series) 

    total_data = { 
        "totalRidershipHistory": condensed_ridership_series,
        "totalServiceHistory": condensed_service_series,
        "totalRidershipPercentage": total_ridership_percentage,
        "totalServicePercentage": total_service_percentage,
        "totalPassengers": total_passengers,
        "totalTrips": combined_total_trips,
        "totalRoutesCancelled": total_cancelled_routes,
        "totalReducedService": total_reduced_serv_routes,
        "totalIncreasedService": total_increased_serv_routes,
    }
    return total_data
```

Reply:
Hmm, okay, I agree that that's a lot of parameters. It'd be okay to split into a function, but also fine to keep as-is. Thanks for the explanation!